### PR TITLE
[JBIDE-21559] Allow access to specific pod container logs

### PR DESCRIPTION
**Pull request by jcantrill**
cc @fbricon @adietish

### CODE REVIEW

**Comment by scabanovich** [comment]
In ConsoleListener.consolesRemoved(IConsole[] consoles) the line REGISTRY.remove(pod) has to be changed to remove Key object as key in the map.

It may be better to create Key instance right in the constructor of PodLogsJob, and make one field Key key that gives access to pod and containerName.

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/handler/PodLogsHandler.java [hunk 1]
```diff
@@ -59,7 +66,25 @@ private void showLogs(IPod pod, ExecutionEvent event) {
...
-		new PodLogsJob(pod).schedule();
+		Collection<IContainer> containers = pod.getContainers();
+		String containerName = null;
+		if(containers.size() > 1) {
+			List<String> names = containers.stream().map(c->c.getName()).collect(Collectors.toList());
+			Collections.sort(names);
+			ElementListSelectionDialog dialog = new ElementListSelectionDialog(HandlerUtil.getActiveShell(event), new LabelProvider());
+			dialog.setElements(names.toArray());
+			dialog.setTitle("Pod Containers");
+			dialog.setMessage("Which container logs do you need?");
```
Comment (scabanovich) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/handler/PodLogsHandler.java]: Could message be like "Select a container log" ?

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/handler/PodLogsHandler.java [hunk 2]
```diff
@@ -59,7 +66,25 @@ private void showLogs(IPod pod, ExecutionEvent event) {
...
+			dialog.setMultipleSelection(false);
+			int result = dialog.open();
+			if(Window.CANCEL == result) {
+				return;
+			}
```
Comment (scabanovich) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/handler/PodLogsHandler.java]: Can user have an option to decline container selection but not to cancel the job and continue as in the case containers.size() == 0 ?

**Comment by jcantrill** [comment]
@scabanovich updated pr per your suggested changes

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/handler/PodLogsHandler.java [hunk 3]
```diff
@@ -59,7 +66,25 @@ private void showLogs(IPod pod, ExecutionEvent event) {
...
+			containerName = (String) dialog.getFirstResult();
+		}else if (containers.size() == 1){
+			containerName = containers.iterator().next().getName();
+		}
+		new PodLogsJob(pod, containerName).schedule();
```
Comment (fbricon) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/handler/PodLogsHandler.java]: technically if containers.size() == 0, then containerName == null, so no need to schedule the job

Comment (jcantrill) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/handler/PodLogsHandler.java]: @fbricon added check for no container though I would imagine the server won't allow this case.

**Comment by fbricon** [comment]
+1 to apply once https://github.com/openshift/openshift-restclient-java/pull/127 is merged and the client pushed to nexus

**Comment by jcantrill** [comment]
merged.

**Pull request closed without merging**